### ISSUE TITLE
Use `MaybeUninit` for newly allocated samples

### ISSUE THREAD

**Issue by elBoberido** [eclipse-iceoryx/iceoryx-rs#1]
Since the data of the samples is not initialized, this should be reflected by the type system with `MaybeUninit`. The user then either has to use `assume_init` before sending the sample or it might be done under the hood. Alternatively `MaybeUninit<T>` could be transmuted to `T`.

**Comment by neilisaac** [comment]
Exposing &mut [u8] is also nice for compatibility with the bytes::BufMut trait. 

For my use case I decided to build a fairly thin rust interface on top of the C bindings and am fairly happy with that approach so far. Happy to compare notes if you’re interested.

**Comment by elBoberido** [comment]
@neilisaac sorry for the late response. I was quite busy with the iceoryx v2 release.

Good idea. I'll add this to the API.

Did you use the current API? If yes, how does it feel? Would be great to get some feedback.

**Comment by neilisaac** [comment]
@elBoberido I found the C API fairly easy to use to build RAII abstractions for each entity.  I built bindings for all C methods exposed but haven't integrated it with a production codebase (yet).  To avoid adding noise to this issue, here's an [example](https://gist.github.com/neilisaac/7427ef223b4e91dae8bb6142b30b8a10) use of my rust bindings.  I'm happy to chat if you're interested in comparing notes about the two approaches for rust support.
